@AssertTrue is not working on ConfigMapping when it returns true

I have the following class

```java
@ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
@ConfigMapping(prefix = "project")
public interface ProjectConfig {
     @AssertTrue(message = "ProjectConfig values are invalid")
    default boolean isValid() {
        return true;
    }
}
```
When `isValid` method returns `true`, I get the following exception:

```java
ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:121)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:77)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:48)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:137)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:116)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: jakarta.validation.ValidationException: HV000041: Call to TraversableResolver.isReachable() threw an exception.
	at org.hibernate.validator.internal.engine.ValidatorImpl.isReachable(ValidatorImpl.java:1316)
	at org.hibernate.validator.internal.engine.ValidatorImpl.isValidationRequired(ValidatorImpl.java:1291)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateMetaConstraint(ValidatorImpl.java:549)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForSingleDefaultGroupElement(ValidatorImpl.java:518)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForDefaultGroup(ValidatorImpl.java:488)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForCurrentGroup(ValidatorImpl.java:450)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:400)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validate(ValidatorImpl.java:172)
	at io.smallrye.config.validator.BeanValidationConfigValidator.validateMappingClass(BeanValidationConfigValidator.java:239)
	at io.smallrye.config.validator.BeanValidationConfigValidator.validateMappingInterface(BeanValidationConfigValidator.java:61)
	at io.smallrye.config.validator.BeanValidationConfigValidator.validateMapping(BeanValidationConfigValidator.java:40)
	at io.smallrye.config.ConfigMappingContext.constructMapping(ConfigMappingContext.java:76)
	at io.smallrye.config.ConfigMappingContext.<init>(ConfigMappingContext.java:58)
	at io.smallrye.config.SmallRyeConfig$1.get(SmallRyeConfig.java:129)
	at io.smallrye.config.SmallRyeConfig$1.get(SmallRyeConfig.java:126)
	at io.smallrye.config.SecretKeys.doUnlocked(SecretKeys.java:28)
	at io.smallrye.config.SmallRyeConfig.buildMappings(SmallRyeConfig.java:126)
	at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:93)
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:773)
	at io.quarkus.runtime.generated.Config.readConfig(Unknown Source)
	at io.quarkus.runtime.generated.Config.createRunTimeConfig(Unknown Source)
	at io.quarkus.deployment.steps.RuntimeConfigSetup.deploy(Unknown Source)
	... 11 more
Caused by: java.lang.IllegalStateException: Persistence providers are not available during the static init phase.
	at io.quarkus.hibernate.orm.runtime.StaticInitHibernatePersistenceProviderResolver.getPersistenceProviders(StaticInitHibernatePersistenceProviderResolver.java:15)
	at jakarta.persistence.Persistence$PersistenceUtilImpl.isLoaded(Persistence.java:142)
	at org.hibernate.validator.internal.engine.resolver.JPATraversableResolver.isReachable(JPATraversableResolver.java:52)
	at org.hibernate.validator.internal.engine.resolver.CachingJPATraversableResolverForSingleValidation.lambda$isReachable$0(CachingJPATraversableResolverForSingleValidation.java:43)
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1228)
	at org.hibernate.validator.internal.engine.resolver.CachingJPATraversableResolverForSingleValidation.isReachable(CachingJPATraversableResolverForSingleValidation.java:43)
	at org.hibernate.validator.internal.engine.ValidatorImpl.isReachable(ValidatorImpl.java:1307)
	... 32 more
```

When `isValid` method returns `false`, I get the following exception:

```java
ERROR [io.qua.run.Quarkus] (Quarkus Main Thread) Error running Quarkus: java.lang.ExceptionInInitializerError
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
	at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:501)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:76)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:48)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:137)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:116)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: io.smallrye.config.ConfigValidationException: Configuration validation failed:
	project valid ProjectConfig.onboarding config values are invalid
	at io.smallrye.config.SmallRyeConfig.buildMappings(SmallRyeConfig.java:139)
	at io.smallrye.config.SmallRyeConfig.<init>(SmallRyeConfig.java:93)
	at io.smallrye.config.SmallRyeConfigBuilder.build(SmallRyeConfigBuilder.java:773)
	at io.quarkus.runtime.generated.Config.<clinit>(Unknown Source)
	... 11 more
```

What I want to achieve is cross validation. If there is a better way to do it, please let me know. I want to validate the config before the application starts (or at least before running any user code, like DB migrations via Flyway, etc)

--- a/extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/HibernateBeanValidationConfigValidator.java
+++ b/extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/HibernateBeanValidationConfigValidator.java
@@ -24,7 +24,8 @@ public class HibernateBeanValidationConfigValidator implements BeanValidationCon
                 .ignoreXmlConfiguration()
                 .builtinConstraints(constraints)
                 .initializeBeanMetaData(classesToBeValidated)
-                .constraintValidatorFactory(new ConstraintValidatorFactoryImpl());
+                .constraintValidatorFactory(new ConstraintValidatorFactoryImpl())
+                .traversableResolver(new TraverseAllTraversableResolver());
 
         ConfigValidatorHolder.initialize(configuration.buildValidatorFactory());
     }

